src/main/java/bio/terra/service/dataset/dao/DatasetDao.java

```diff
@@ -373,6 +387,7 @@ public boolean deleteByNameAndFlight(String datasetName, String flightId) {
         return rowsAffected > 0;
     }
 
+    @Transactional(propagation = Propagation.SUPPORTS, readOnly = true)
```

Comment:
Any reason these getters aren't `Propagation.REQUIRED`?...I don't think we want these non-transactional ever since  `retrieveWorker` reaches across DAOs